packages/EasyTest/src/Traits/MessengerAssertionsTrait.php

```diff
@@ -206,8 +220,58 @@ private static function runMessengerWorker(string $transportName, string $busNam
             $messageLimitListener = new StopWorkerOnMessageLimitListener($messagesCount);
             $eventDispatcher->addSubscriber($messageLimitListener);
 
-            $worker = new Worker(['async' => $asyncTransport], $messageBus, $eventDispatcher);
+            $worker = new Worker(['async' => $transport], $messageBus, $eventDispatcher);
             $worker->run();
         }
+
+        if (\count((array)$transport->get()) > 0) {
+            throw new \RuntimeException('Unable to consume all messages from async transport.');
```

Comment:
What case does this cover?

Reply:
This is a case where one message triggers other messages, and those messages trigger more messages, and so on.

Here we check that we have processed all the messages in the test.

Reply:
OK, thanks. But what should we do then if some message hadler dispatched another message?

Reply:
Explanation added in doc comments.